src/coreclr/jit/valuenum.cpp

```diff
@@ -15151,3 +15190,103 @@ void ValueNumStore::PeelOffsets(ValueNum* vn, target_ssize_t* offset)
         }
     }
 }
+
+//--------------------------------------------------------------------------------
+// VNUnwrapPhis: given a VN, call the specified callback function on it. In case if the VN
+//    is a phi node, recursively call the callback function on all the VN arguments of that phi node.
+//
+// Arguments:
+//    vn         - The VN to unwrap
+//    argVisitor - The callback function to call on the vn and its PHI arguments if any
+//    maxDepth   - Maximum depth to handle recursive PHI chains. -1 means no limits.
+//    vnk        - The kind of VN to use (Conservative or Liberal)
+//
+// Return Value:
+//    UnwrapResult::MaxDepthReached - we hit the maxDepth limit (for TP reasons)
+//    UnwrapResult::Aborted         - any argVisitor returned Unwrap::Abort
+//    UnwrapResult::Completed       - unwrap was successful (all argVisitor returned Unwrap::Continue)
+//
+template <typename TArgVisitor>
+ValueNumStore::UnwrapResult ValueNumStore::VNUnwrapPhis(ValueNum     vn,
```

Comment:
Sorry to nitpick on the naming...

I liked something with "Visit" or "Walk" in the name. It follows same conventions as many other functions we have of this pattern in the JIT.

Andy calls the thing we are visiting here the SSA web (and searching around, that seems to be used in more places). So I would suggest `VisitSSAWeb`.

Reply:
Hmm, the SSA web would probably also include the uses of all the phi defs we see. But this is more like visiting all the possible VNs that this VN may take. Perhaps something like `VisitReachingVNs` would make sense then, to make the correspondence with reaching defs clear.

Reply:
`VisitReachingVNs` sounds fine. I wasn't happy with Unwrap, but I didn't like the previous WalkPhi because the the function has changed - it no longer gives up on non-phi nodes